storage/dataflux/fast_list.go

```diff
@@ -150,6 +179,39 @@ func (c *Lister) NextBatch(ctx context.Context) ([]*storage.ObjectAttrs, error)
 
 // Close closes the range channel of the Lister.
 func (c *Lister) Close() {
+	if c.ranges != nil {
+		close(c.ranges)
+	}
+}
 
-	// TODO: Close range channel for worksteal lister.
+// updateStartEndOffset updates start and end offset based on prefix.
+// If a prefix is given, start and end should contain the value after prefix.
+// For example, if start is "prefix_a", then start should be "_a".
```

Comment:
What happens if a prefix is given, ex "someprefix", but `start`/`end` does not contain that prefix? Looks like based on my understanding of the code below, if `start` is lexicographically smaller than "someprefix", it would return the empty string, otherwise it would just return `start`? 

Can you update this comment with the expected behaviour?
Also please add a unit test for this method.

Reply:
This is what will be  expected.  
If a `prefix` is given and `start` is smaller, that means listing will begin from the very first object with the given `prefix`. So the smallest possible `Startoffset` is the prefix. 

I have added an example.